api/src/main/java/org/apache/iceberg/PartitionField.java

```diff
@@ -70,14 +100,14 @@ public boolean equals(Object other) {
     }
 
     PartitionField that = (PartitionField) other;
-    return sourceId == that.sourceId
+    return sourceIds.toString().equals(that.sourceIds.toString())
```

Comment:
Just wondering why not `Objects.equals(sourceIds, that.sourceIds);` do we need the extra string conversion?

Reply:
Yeah, either way is fine. Let me do the change if you prefer.